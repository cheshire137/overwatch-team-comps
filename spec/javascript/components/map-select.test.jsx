import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import MapSelect from '../../../app/assets/javascripts/components/map-select.jsx'

describe('MapSelect', () => {
  let component = null
  const map1 = {
    id: 4,
    slug: 'dorado',
    name: 'Dorado',
    type: 'escort',
    segments: [{ id: 31, name: 'Attack: Payload 1' }],
    compositions: []
  }
  let mapIDSelected = map1.id
  const map2 = {
    id: 7,
    slug: 'eichenwalde',
    name: 'Eichenwalde',
    type: 'hybrid',
    segments: [{ id: 61, name: 'Attack: Point 1' }],
    compositions: []
  }
  const props = {
    maps: [map1, map2],
    onChange: newMapID => { mapIDSelected = newMapID },
    selectedMapID: mapIDSelected,
    disabled: false
  }

  beforeEach(() => {
    component = <MapSelect {...props} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('can change selected map', () => {
    const rendered = shallow(component)

    const select = rendered.find('.menu-toggle')
    select.simulate('click', { target: { blur: () => {} } })

    const newMapButton = rendered.find('.menu-item.map-eichenwalde')
    newMapButton.simulate('click', {
      preventDefault: () => {},
      target: { blur: () => {} }
    })

    expect(mapIDSelected).toBe(map2.id)
  })
})
