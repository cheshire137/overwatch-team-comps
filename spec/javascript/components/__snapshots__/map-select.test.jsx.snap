exports[`MapSelect matches snapshot 1`] = `
<div
  className="map-select-container menu-container">
  <button
    className="button menu-toggle "
    disabled={false}
    onClick={[Function]}
    type="button">
    <i
      aria-hidden="true"
      className="fa fa-map-marker" />
     
    Dorado
     
    <i
      aria-hidden="true"
      className="fa fa-caret-down" />
  </button>
  <div
    className="menu">
    <button
      className="map-dorado menu-item button is-selected"
      onClick={[Function]}>
      Dorado
      <i
        aria-hidden="true"
        className="fa fa-check menu-item-selected-indicator" />
    </button>
    <button
      className="map-eichenwalde menu-item button "
      onClick={[Function]}>
      Eichenwalde
      
    </button>
  </div>
</div>
`;
